pages/search.vue

```diff
@@ -0,0 +1,162 @@
+<template>
+  <div>
+    <v-app-bar fixed flat dense class="second-toolbar">
+      <v-tabs v-model="searchTab" centered>
+        <v-tab :key="0">{{ $t('search.topResults') }}</v-tab>
+        <v-tab :key="1">{{ $t('movies') }}</v-tab>
+        <v-tab :key="2">{{ $t('shows') }}</v-tab>
+        <v-tab :key="3">{{ $t('albums') }}</v-tab>
+        <v-tab :key="4">{{ $t('songs') }}</v-tab>
+        <v-tab :key="5">{{ $t('books') }}</v-tab>
+        <v-tab :key="6">{{ $t('people') }}</v-tab>
+        <v-tab :key="7">{{ $t('artists') }}</v-tab>
+      </v-tabs>
+    </v-app-bar>
+    <v-container class="after-second-toolbar">
+      <v-row>
+        <v-col>
+          <v-tabs-items v-model="searchTab" class="transparent">
+            <v-tab-item :key="0">
+              <item-grid :items="topSearchResults" :loading="loading" />
+            </v-tab-item>
+            <v-tab-item :key="1">
+              <item-grid :items="movieSearchResults" :loading="loading" />
+            </v-tab-item>
+            <v-tab-item :key="2">
+              <item-grid :items="showSearchResults" :loading="loading" />
+            </v-tab-item>
+            <v-tab-item :key="3">
+              <item-grid :items="albumSearchResults" :loading="loading" />
+            </v-tab-item>
+            <v-tab-item :key="4">
+              <item-grid :items="trackSearchResults" :loading="loading" />
+            </v-tab-item>
+            <v-tab-item :key="5">
+              <item-grid :items="bookSearchResults" :loading="loading" />
+            </v-tab-item>
+            <v-tab-item :key="6">
+              <item-grid :items="personSearchResults" :loading="loading" />
+            </v-tab-item>
+            <v-tab-item :key="7">
+              <item-grid :items="artistSearchResults" :loading="loading" />
+            </v-tab-item>
+          </v-tabs-items>
+        </v-col>
+      </v-row>
+    </v-container>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { BaseItemDto } from '@jellyfin/client-axios';
+
+export default Vue.extend({
+  data() {
+    return {
+      loading: false,
+      searchTab: 0,
+      topSearchResults: [] as BaseItemDto[],
+      movieSearchResults: [] as BaseItemDto[],
+      showSearchResults: [] as BaseItemDto[],
+      albumSearchResults: [] as BaseItemDto[],
+      trackSearchResults: [] as BaseItemDto[],
+      bookSearchResults: [] as BaseItemDto[],
+      personSearchResults: [] as BaseItemDto[],
+      artistSearchResults: [] as BaseItemDto[]
+    };
+  },
+  computed: {
+    searchQuery(): string {
+      return this.$route.query?.q || '';
+    }
+  },
+  watch: {
+    searchQuery(newQuery: string, _oldQuery: string): void {
+      if (newQuery === '') {
+        this.$router.back();
+      } else {
+        this.performSearch();
+      }
+    }
+  },
+  beforeMount() {
+    if (this.searchQuery === '') {
+      this.$router.back();
```

Comment:
Same here?

Reply:
I think here we still want to go to back if something has gone wrong, and the query is ''

Reply:
Oh yeah didn't think about the `beforeMount`, np then :)

Reply:
Okay yeah I see it, it's good then :)